src/opnsense/scripts/unbound/start.sh

```diff
@@ -26,11 +26,26 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-set -e
-
 # prepare and startup unbound, so we can easily background it
 
-chroot -u unbound -g unbound / /usr/local/sbin/unbound-anchor -a /var/unbound/root.key
+# if the root.key file is missing or damaged, run unbound-anchor
+/usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null 
+
+if [ $? -eq 1 ]; then
```

Comment:
```suggestion
if ! /usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null; then
```

assuming "1" is returned is strange for all sorts of error behaviour that could be embedded here or later.. better to assume ! 0 and avoid going through return variable if we don't need it again

Reply:
Fair enough, much better.